etna/transforms/timestamp/holiday.py

```diff
@@ -60,19 +91,42 @@ def _get_column_name(self) -> str:
             return self.__repr__()
 
     def _fit(self, df: pd.DataFrame) -> "HolidayTransform":
+        """Fit the transform.
+
+        Parameters
+        ----------
+        ts:
+            Dataset to fit the transform on.
+
+        Returns
+        -------
+        :
+            The fitted transform instance.
         """
-        Fit HolidayTransform with data from df. Does nothing in this case.
+        return self
+
+    def fit(self, ts: TSDataset):
+        """Fit the transform.
 
         Parameters
         ----------
-        df: pd.DataFrame
-            value series with index column in timestamp format
+        ts:
+            Dataset to fit the transform on.
+
+        Returns
+        -------
+        :
+            The fitted transform instance.
         """
+        df = ts.to_pandas(flatten=False, features=self.required_features)
+        self._fit(df=df)
+        self.freq = ts.freq
         return self
 
     def _transform(self, df: pd.DataFrame) -> pd.DataFrame:
         """
         Transform data from df with HolidayTransform and generate a column of holidays flags or its titles.
+        In days_count mode calculates the frequency of holidays in a given period of time
```

Comment:
We should add "Raises block" in docstring like here: https://github.com/etna-team/etna/blob/71b80d2d7e1ec0f44391f9a7768435da6f0ab9cd/etna/transforms/decomposition/deseasonal.py#L88